Updates to have oslo_aero work in the noresm_develop version of CAM

mvertens opened the pull request:
- changed interface for call to qsat_water 
- removed references to SPMD
- added logic to have DMS flux from ocean use the nuopc cap
- changed following namelists to have additional namelist variables generated by cam - but not used by oslo_aero for 
    -  hetfrz_classnuc_nl: hetfrz_bc_scalfac, hetfrz_dust_scalfac)
    - microp_aero_nl:  microp_aero_wsub_min,  microp_aero_wsubi_min, microp_aero_wsub_scale,microp_aero_wsubi_scale
    - aerosol_nl:  sol_factb_interstitial, sol_factic_interstitial,  modal_strat_sulfate, modal_accum_coarse_exch, seasalt_emis_scale
- fixed ncol / pcol errors
- fixed missing lchnk error

mvertens committed:
changes to get noresm_develop working with oslo_aero

mvertens committed:
removed SPMD and fixed logic for having dms flux come from ocean

mvertens committed:
updates to clean up interfaces

mvertens committed:
updates to be consistent with changes in radiation.F90

mvertens committed:
removed state input to oslo_aero_depos

mvertens committed:
more refactoring to remove cam data types

mvertens committed:
more cleanup of state% variables

mvertens committed:
updates go get AEROCOM working

mvertens committed:
fixed problems with usage of nmodes

mvertens committed:
more cleanup

mvertens committed:
changes to get ice nucleation working correctly with cam_dev and also backing out updates in oslo_aero_optical_params to be in line with release implementation

gold2718 reviewed (approved):
This is a really nice piece of work.
The refactor makes it hard to do a detailed code review, however, once the simulation is validated, it will be a really good place to begin the modularization into the new objects.

mvertens committed:
simplicification of oslo_aero_optical_params

mvertens committed:
added shadow files with OSLO_AERO ifdefs to new src_cam directory

mvertens committed:
simplified calls to oslo_aero_optical_params from radation.F90

mvertens committed:
renamed oslo_model_aero to model_aero

TomasTorsvik commented:
Sorry, not my turf, but why is this PR marked as "done" when it is not merged?

gold2718 commented:
> Sorry, not my turf, but why is this PR marked as "done" when it is not merged?

Good question, I'm not sure how (or when) that happened. I put it back to "In Progress", thanks for pointing it out.

mvertens committed:
fixes to get PEM test working

Commit:
Fix chunk / ncol mismatch errors
Fix several outfld pcols / ncol errors, AEROCOM still not finished
Typos, cleanup, clothe some naked use statements

mvertens committed:
Merge pull request #1 from gold2718/oslo_aero_integrate

Fix chunk / ncol mismatch errors
